src/test/java/org/example/FilteringSearching.java

```diff
@@ -1,5 +1,147 @@
 package org.example;
 
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+
 public class FilteringSearching {
+    private WebDriver driver;
+    private static final String AMAZON_URL = "https://www.amazon.com/";
+    private static final Duration TIMEOUT = Duration.ofSeconds(10);
+    private final By categories = By.xpath("//h2[text()='Shop by Category' or text()='Gaming accessories']/../following-sibling::div//div[contains(@class, 'quadrant-container')] | //img[@alt='Electronics']");
+    private final By featuredBrands = By.xpath("//span[text()='Featured Brands' or text()='Brands']/../..//span[@class='a-size-base a-color-base' and not(ancestor::div[@class='a-row a-expander-container a-expander-extend-container'])]");
+    private final By searchResults = By.xpath("//span[@data-component-type='s-search-results' and not(contains(@class, 'AdHolder'))]//span[@class='a-size-base-plus a-color-base a-text-normal' or @class='a-size-medium a-color-base a-text-normal']");
+    private final By minField = By.id("low-price");
+    private final By maxField = By.id("high-price");
+    private final By goBtn = By.xpath("//input[@class='a-button-input']");
+    private final By priceResults = By.xpath("//div[@data-component-type='s-search-result']//span[@data-a-color='base']/span[@class='a-offscreen']");
+    private final By checkedCheckbox = By.xpath("//input[@checked]/../../following-sibling::span");
+    private final By sortByDropDown = By.xpath("//span[@data-csa-c-func-deps='aui-da-a-dropdown-button']");
+    private final By lowToHigh = By.id("s-result-sort-select_1");
+    private final By highToLow = By.id("s-result-sort-select_2");
+
+
+    @BeforeClass
+    private void setDriver() {
+        System.setProperty("webdriver.chrome.driver", "src/test/resources/webdriver/chromedriver.exe");
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+    }
+
+    @AfterClass
+    private void closeDriver() {
+        driver.quit();
+    }
+
+    @Test
+    public void testSearchingBrandedItems() {
+        openPage(AMAZON_URL);
+        clickRandomElement(categories);
+        clickRandomElement(featuredBrands);
```

Comment:
Since every test contains these three steps at the beginning, you can add them to the beforeclass or create a new @BeforeTest method